src/sdc11073/consumer/serviceclients/setservice.py

```diff
@@ -40,7 +40,7 @@ def set_numeric_value(self, operation_handle: str,
                           operation_handle, requested_numeric_value)
         request = data_model.msg_types.SetValue()
         request.OperationHandleRef = operation_handle
-        request.RequestedNumericValue = requested_numeric_value
+        request.RequestedNumericValue = Decimal(requested_numeric_value)
```

Comment:
the problem is, that if you input a float value, e.g. `decimal.Decimal(1.1)` the resulting decimal will be `Decimal('1.100000000000000088817841970012523233890533447265625')` due top floating point conversion. In order to fix this, you first need to cast it to a `str` and then to a `Decimal`. I suppose we restrict this call to `Decimal` only type, or we cast it every time to a `str` first.

Reply:
don't want to change the interface, implemented str conversion proposal.